File: lang/php/src/vac.fun/algorithms/number_theory/Coprimality.php
<?php
    namespace vac\fun\algorithms\number_theory\Coprimality;

    require_once "vac.fun/algorithms/number_theory/GreatestCommonDivisor.php";

    use vac\fun\algorithms\number_theory\GreatestCommonDivisor;

    function reduceToBinaryGCD(int $m, int $n): bool {
        return GreatestCommonDivisor\iterativeBinaryGCD($m, $n) == 1;
    }

    function reduceToEuclidean(int $m, int $n): bool {
        return GreatestCommonDivisor\iterativeEuclidean($m, $n) == 1;
    }
?>

File: rust/muse/src/algorithms/number_theory/coprimality.rs
use super::greatest_common_divisor;

pub fn reduce_to_binary_gcd(m: i64, n: i64) -> bool {
    greatest_common_divisor::iterative_binary_gcd(m, n) == 1
}

pub fn reduce_to_euclidean(m: i64, n: i64) -> bool {
    greatest_common_divisor::iterative_euclidean(m, n) == 1
}

#[cfg(test)]
pub(crate) mod tests;
